### ISSUE TITLE
Shell provisioners in Ubuntu 16.04 still output "mesg: ttyname failed : Inappropriate ioctl for device"

### ISSUE THREAD

**Issue by dominics** [mitchellh/vagrant#8288]
### Vagrant version

Vagrant 1.9.1 - but this error has been present throughout 1.8.x and 1.9.x

### Host operating system

Mac OS X personally, but this problem has been confirmed on Linux hosts too

### Guest operating system

Ubuntu 16.04 (bento, canonical, etc.)

### Vagrantfile

```ruby
Vagrant.configure(2) do |config|
  config.vm.box = "bento/ubuntu-16.04"
  config.vm.provision "shell", inline: "echo 'Hello, world!'"
end
```

### Expected behavior

```
==> default: Running provisioner: shell...
    default: Running: inline script
==> default: Hello, world!
```

### Actual behavior

```
==> default: Running provisioner: shell...
    default: Running: inline script
==> default: mesg:
==> default: ttyname failed
==> default: :
==> default: Inappropriate ioctl for device
==> default: Hello, world!
```

### References

- #7368 - Seems to be a straightforward duplicate of this bug that has been closed since December, after #8051 was applied. But that PR has not fixed the issue. Feel free to close this issue if work should resume on #7368.

**Comment by chrisroberts** [comment]
@dominics Is this using the virtualbox provider?

**Comment by richard-scott** [comment]
config.vm.provision "fix-no-tty", type: "shell" do |s|
    s.privileged = true
    s.inline = "sed -i '/tty/!s/mesg n/tty -s \\&\\& mesg n \\|\\| true/' /r$  end

**Comment by dominics** [comment]
A big thank you @chrisroberts

**Comment by lindes** [comment]
See also: https://superuser.com/a/1277604/57367
